### ISSUE TITLE
Hydration issues - Nuxt 3/Vue 3

### ISSUE THREAD

**Issue by jawngee** [Akryum/floating-vue#1006]
This just started showing up in the latest Nuxt 3.9, probably due to the Vue version bump but I'm not sure.

Getting a lot of Hydration attribute mismatch because the ID of the popper is different from SSR.  Errors look like:

```
- rendered on server: id=\"popper_44tr48ae_tdggwc\"
- expected on client: id=\"popper_gjw3s4ml_tdglzw\"
  Note: this mismatch is check-only. The DOM will not be rectified in production due to performance overhead.
```

Not sure how to solve this or if it has any real impact at all.

**Comment by simba77** [comment]
As a temporary solution, you can add the **aria-id** attribute
`<v-dropdown aria-id="some-dropdown">`

**Comment by Akryum** [comment]
Yes you can use [ariaId](https://floating-vue.starpad.dev/api/#ariaid) prop to prevent using a random id for the `aria-describedby` attribute.
Maybe it could be fixed in nuxt via the module but I'm not sure.

**Comment by Akryum** [comment]
Also is this caused by popper that are already open when they are rendered on the server? For example with `:shown="true"`

**Comment by jawngee** [comment]
Thanks, I will give the aria thing a try.

No, they aren't opened but let me double check that t confirm.

**Comment by sbittmann** [comment]
in Nuxt3 I used
`<ClientOnly>
    <template #fallback>
        <Item2Render />
    </template><Item2Render /><template #popper<PopperContent /></template><ClientOnly>`

**Comment by yoshrubin** [comment]
the above works for the `ariaId` issue but I am getting 
```
 - rendered on server: tabindex=""
  - expected on client: tabindex="0"
  Note: this mismatch is check-only. The DOM will not be rectified in production due to performance overhead.
  You should fix the source of the mismatch. 
  at <VPopperContent ref="popperContent" popper-id="dropdown" theme="info-dropdown"  ... > 
  at <Popper ref="popper" theme="info-dropdown" referenceNode=null  ... > 
  at <Dropdown triggers= ['click'] skidding=-30 distance=25 > 
  ```
  when using a `Dropdown` so the contents of the `Popper` are clickable

**Comment by Mini-ghost** [comment]
> Yes you can use [ariaId](https://floating-vue.starpad.dev/api/#ariaid) prop to prevent using a random id for the `aria-describedby` attribute. Maybe it could be fixed in nuxt via the module but I'm not sure.

This was useful for me, but it might not be the best approach as it causes the vuejs-accessibility's aria-props to throw errors at me. I wonder if there's another way to address this?


<img width="800" alt="eslint-disable vuejs-accessibility/aria-props" src="https://github.com/Akryum/floating-vue/assets/39984251/9b7e08c4-b7d8-475b-9182-ca1904f7a434">

**Comment by max13h** [comment]
> the above works for the `ariaId` issue but I am getting
> 
> ```
>  - rendered on server: tabindex=""
>   - expected on client: tabindex="0"
>   Note: this mismatch is check-only. The DOM will not be rectified in production due to performance overhead.
>   You should fix the source of the mismatch. 
>   at <VPopperContent ref="popperContent" popper-id="dropdown" theme="info-dropdown"  ... > 
>   at <Popper ref="popper" theme="info-dropdown" referenceNode=null  ... > 
>   at <Dropdown triggers= ['click'] skidding=-30 distance=25 > 
> ```
> 
> when using a `Dropdown` so the contents of the `Popper` are clickable

Hey, same issue here

**Comment by Dodje** [comment]
I believe it's not a very good thing to generate id for every tooltip and dropdown manually..

**Comment by Nuzzlet** [comment]
[Headless UI solved this problem with the new UseID function in Nuxt](https://github.com/tailwindlabs/headlessui/issues/2913#issuecomment-1924681165)
This solution uses Nuxt's new UseID composable to generate an SSR safe ID.
The warning is caused by the latest version of Nuxt which improved hydration mismatch detection.

**Comment by Nuzzlet** [comment]
My soloution for the time being is this:
`
<script setup>
const ariaId = useId()
</script>
<template>
<VDropdown :aria-id="ariaId"  />
</template>
`

**Comment by TheDogHusky** [comment]
will there be any official fix such as wrapping the whole element where tooltips and dropdowns should go into a ClientOnly or the possibility for us to manually set it up?

**Comment by mrleblanc101** [comment]
@Akryum Can we move to using `useId()` as it's SSR safe instead of `Math.random()` ?

https://github.com/Akryum/floating-vue/blob/19857764c4f73dea7ed44a7d970adb968ee7ad90/packages/floating-vue/src/components/Popper.ts#L317

**Comment by ChrizzDF** [comment]
Yes, if I set `aria-id` the hydration issue is gone but I still get an accessibility warning:

aria-id This attribute is an invalid ARIA attribute.eslint[vuejs-accessibility/aria-props](https://vue-a11y.github.io/eslint-plugin-vuejs-accessibility/rules/aria-props.html) because well, `aria-id` is not a valid `aria-` attribute (https://www.w3.org/TR/wai-aria/#state_prop_def)

Maybe this can be adjusted 🙏

**Comment by mrleblanc101** [comment]
@ChrizzDF it's just the name of the props, so it is valid. It's not rendered in the HTML
